### PR TITLE
Add Contributions USA messaging A/B test

### PR DESCRIPTION
**Pull request by jranks123**
## What does this change?

This add's a new A/B test which tests 5 different messages in contributions asks. Only people with a US IP address will see the test, as we use the new geolocation endpoint in fastly to determine the users location.  

Also, we now check to see if someone has ever contributed before we show them an ad. If they have, we do not show the ad!
## What is the value of this and can you measure success?

We will see what messaging works with a US audience, and we will measure success by comparing the amount impressions to the amount of contributions.

## Does this affect other platforms - Amp, Apps, etc?

No

## Screenshots

usa1
usa2
usa3
usa4
usa5

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        var bottomWriter = function (component) {
+            return fastdom.write(function () {
+                try {
+                    ajax({
+                        url: 'http://api.nextgen.guardianapps.co.uk/geolocation',
+                        method: 'GET',
+                        contentType: 'application/json',
+                        crossOrigin: true
+                    }).then(function (resp) {
+                        if(resp.country == 'US') {
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: `resp.country === 'US'`?

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                            var submetaElement = $('.submeta');
+                            component.insertBefore(submetaElement);
+                            embed.init();
+                            mediator.emit('contributions-embed:insert', component);
+                        }
+                    });
+                } catch (e) {
+                    //Do nothing
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: Are we expecting this to fail?

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    ) {
+
+
+    return function () {
+
+        this.id = 'ContributionsUsa1';
+        this.start = '2016-10-20';
+        this.expiry = '2016-10-26';
+        this.author = 'Jonathan Rankin';
+        this.description = 'Test which of our 5 initial US targetted messages results in the most contributions';
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: ~~targetted~~ targeted 

(pedantic I know 😛 )

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+            },
+
+            {
+                id: 'independent',
+                test: function () {
+                    var component = $.create(template(contributionsEmbed, {
+                        position : 'inline',
+                        variant: 'bottom',
+                        p1: 'No billionaire owner, no shareholders. Just independent, investigative reporting that fights for the truth, whatever the cost. Why not suport it?',
+                        linkUrl: 'https://contribute.theguardian.com/us?INTCMP=co_us_editorial_footer_independent&amount=50',
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: ~~suport~~ support

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+            var userHasNeverContributed = !cookies.get('gu.contributions.contrib-timestamp');
+            var worksWellWithPageTemplate = (config.page.contentType === 'Article'); // may render badly on other types
+            return userHasNeverContributed && commercialFeatures.canReasonablyAskForMoney && worksWellWithPageTemplate;
+        };
+
+        var bottomWriter = function (component) {
+            return fastdom.write(function () {
+                try {
+                    ajax({
+                        url: 'http://api.nextgen.guardianapps.co.uk/geolocation',
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: this should be `guardian.config.page.ajaxUrl + '/geolocation'` I imagine

**Review by jfsoul — CHANGES_REQUESTED** [review]
See comments + discussion in person, but looks good  overall!

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js (hunk 3 above)
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: Done

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js (hunk 2 above)
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: Done

File: static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js (hunk 4 above)
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/contributions-usa-1.js]: Done

**Review by jfsoul — APPROVED** [review]

**Comment by jfsoul** [comment]
Nice work @jranks123 and thanks for listening to my pedantic suggestions! 👏 👍

**Pull request merged**